lib/src/async/future_stream.dart

```diff
@@ -47,7 +47,7 @@ class FutureStream<T> extends Stream<T> {
         _controller.close();
       }
       _controller = null;
-    });
+    }) as Future<Stream<T>>;
```

Comment:
On second thought, this may not need `catchError` at all. This should work just as well:

```dart
_future = future.then((Stream<T> s) => s, onError: (e, stackTrace) {
  // Since [controller] is synchronous, it's likely that emitting an error
  // will cause it to be cancelled before we call close.
  if (_controller != null) {
    _controller.addError(e, stackTrace);
    _controller.close();
  }
  _controller = null;
});
```

Would this pass strong mode?